### ISSUE TITLE
 Tags - The tag custom name is not truncated to one line

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#54415]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.77-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to the tag list in the OldDot policy editor
2. Modify the custom name to be very long (near or at the max of 256 characters)
3. Create a request in NewDot, and select any tag
4. Go to the expense details created in the previous step

## Expected Result:
The tag custom name is truncated to one line in both the header of the row on request detailed as well as the header when clicking in to edit it

## Actual Result:
The tag custom name is not truncated to one line in both the header of the row on request detailed as well as the header when clicking in to edit it

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 
![Bug6680693_1732978308645!not_truncated_2](https://github.com/user-attachments/assets/9608cf5a-1d37-4fa0-b773-dd6939dba21b)

![Bug6680693_1732978308650!not_truncated_2](https://github.com/user-attachments/assets/7ec45371-b7bb-4019-a16f-bcabb894354f)

https://github.com/user-attachments/assets/12a3afc0-5536-4a60-b72d-2a6f3b81a231

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Tags - The tag custom name is not truncated to one line

### What is the root cause of that problem?
1. We a haven't set numberOfLinesDescription to 1 here
https://github.com/Expensify/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/components/ReportActionItem/MoneyRequestView.tsx#L415-L421
2. also in Tag picker page the header default title number of lines is 2
https://github.com/Expensify/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/components/Header.tsx#L72-L75
### What changes do you think we should make in order to solve the problem?

1. set numberOfLinesDescription
```
                    numberOfLinesDescription={1}

```
2. Set an prop of titlenumberOfLines and pass 1 all the way to from IOURequestStepTag to Header and use it [here](https://github.com/Expensify/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/components/Header.tsx#L72-L75)
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by abekkala** [comment]
> Modify the custom name to be very long (near or at the max of 256 characters)

This sounds like an extreme edge case and is most likely pretty low priority - 
@Expensify/design is this something you'd like to keep open at a low priority, or simply close?

**Comment by dannymcclain** [comment]
I agree this seems pretty extreme and will likely never be an actual problem a customer experiences. ...But if a customer was to set a super long custom name, would we _want_ to truncate it? 🤷‍♂️

I'd be fine closing this one, but I'll let @shawnborton and @dubielzyk-expensify weigh in too.

**Comment by shawnborton** [comment]
I think we'd want to truncate it, yes. Though I agree, seems kinda nuts to have a tag name that long. @trjExpensify or @JmillsExpensify any idea why we don't force a reasonable character limit on the tag name?

**Comment by dannymcclain** [comment]
>I think we'd want to truncate it, yes.

But isn't it the label for the push input? Feels kinda weird to truncate a label. I'd be happier putting a lower character limit on it, but I honestly don't feel strongly because this doesn't feel like a super real situation to me. Just `:my2cents:`

**Comment by shawnborton** [comment]
But do we have examples of two-line labels for our push inputs? I feel like most are just one line. I don't feel strongly and I'm happy to do nothing here, but I think my worry is that this mock here:
![CleanShot 2024-12-20 at 16 19 43@2x](https://github.com/user-attachments/assets/93b3c139-2675-4fcf-91cb-754eaf1d0195)

...on mobile probably looks insane if we don't truncate?

**Comment by dannymcclain** [comment]
Haha yeah that is a good point. I bet it does look insane.

**Comment by JmillsExpensify** [comment]
Interesting. I kind of feel like this wouldn't happen in practice. I'm also a bit nervous to implement a limit given that these names can come from accounting integrations.
